### PR TITLE
fix(deploy): deselect removed nodes before synchronization

### PR DESCRIPTION
**Pull request by devantler**
> 🤖 Generated by the Agentic Engineer

## Summary

- Deselect a selected Talos node only when its first pre-mutation read proves absence and a fresh, validated inventory confirms that no replacement reuses its name, UID, or address.
- Keep API errors, ambiguous inventories, bootstrap-owned targets, and post-mutation disappearance fail-closed. A removed target receives no mutation or verification credit, and removing every required target cannot produce an empty successful rollout.
- Continue the bounded convergence pass for remaining and newly selected nodes, retaining the existing two-clean-inventory completion requirement.

Part of #3186. Related to the failed merge-group deployment on #3584. This is the independently testable pre-mutation-removal slice; handling a still-present deleting node with an owned fence remains outside this PR, so #3186 stays open.

## Verification

- RED: the new removed-target convergence test failed on the original script's missing-node error after an earlier target completed.
- GREEN: the full Go suite passed; the final uncached rollout package passed again after adding bootstrap recovery coverage.
- Negative cases cover failed/forbidden reads, failed/malformed/empty inventory, identity or address reuse, every target removed, post-mutation removal, and removal of a bootstrap-owned seed with durable recovery retained.
- `go vet ./...`, `shellcheck -x scripts/refresh-flux-ghcr-auth.sh`, and `git diff --check` passed.

Tests use the existing fake-command harness. No live deployment, merge-queue retry, credential rotation, or cluster mutation was performed for this change.

### CODE REVIEW

File: scripts/tests/refresh-flux-ghcr-auth/rollout_convergence_test.go [hunk 1]
```diff
@@ -208,6 +208,115 @@ func TestFailedImageOnlyPullKeepsNodeCordoned(t *testing.T) {
...
+			t.Parallel()
+			f := newFixture(t)
+			result := f.runHelper(validConfig(), nil, map[string]string{
+				"FAKE_TALOS_NODES_CURRENT":         "true",
+				"FAKE_TALOS_VERIFIED_IMAGE":        "ghcr.io/devantler-tech/ksail:v7.166.0",
+				"FAKE_NODE_REMOVED_BEFORE_PROCESS": "prod-worker-1",
+				"FAKE_REMOVAL_CONFIRMATION":        mode,
+			})
+			requireFailureResult(t, result)
+			operations := readLines(f.operationLog)
```
[the remark this answers is not in the document]
Comment (devantler) [reply scripts/tests/refresh-flux-ghcr-auth/rollout_convergence_test.go:277]: > 🤖 Generated by the Agentic Engineer

Refuted on current head `0125d0e97b954cacf532ef869798258f024aca07`: the exact `TestRemovedTargetRequiresAuthoritativeUnambiguousInventory` table, including both `forbidden` and `same-name`, passed without an `ENOENT` panic, and the uncached full rollout package passed in 177.172s. The operation log is created by earlier fixture activity in these paths, so the claimed absent-file state is not reachable in this test. No production or test-helper change was made for this finding.

File: scripts/refresh-flux-ghcr-auth.sh [hunk 2]
```diff
@@ -2976,8 +3000,17 @@ process_talos_node_target() {
...
   fi
+  # Bootstrap preparation can already own a durable fence on this target.
+  # Its recovery remains fail-closed; only an untouched target may be skipped.
+  if [[ -f "${bootstrap_state_file}" ]]; then
+    allow_removed=0
+  fi
   revalidate_selected_node_identity_before_mutation \
-    "${node_name}" "${node_uid}" "${node_ip}" "${node_role}" || return 1
+    "${node_name}" "${node_uid}" "${node_ip}" "${node_role}" \
+    "${allow_removed}" || identity_result=$?
```
[the remark this answers is not in the document]
Comment (devantler) [reply scripts/refresh-flux-ghcr-auth.sh:3011]: > 🤖 Generated by the Agentic Engineer

Fixed on current head `0125d0e97b954cacf532ef869798258f024aca07`. Pending reboot targets are now revalidated into a separate batch before peer-overlap or bootstrap selection; only authoritative absence with no name, UID, or address replacement is deselected, while later mutation-boundary validation remains. The new `TestRemovedRebootTargetIsDeselectedBeforeBootstrapSelection` failed RED at bootstrap candidate inspection, then passed GREEN; the uncached full rollout package passed in 177.172s, with `go vet ./...`, shellcheck, and diff checks clean.

File: scripts/tests/refresh-flux-ghcr-auth/rollout_convergence_test.go [hunk 3]
```diff
@@ -208,6 +208,142 @@ func TestFailedImageOnlyPullKeepsNodeCordoned(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (devantler) [reply scripts/tests/refresh-flux-ghcr-auth/rollout_convergence_test.go:277]: > 🤖 Generated by the Agentic Engineer

I could not reproduce this finding at current head `0125d0e97b954cacf532ef869798258f024aca07`. The exact command `go test ./scripts/tests/refresh-flux-ghcr-auth -run '^TestRemovedTargetRequiresAuthoritativeUnambiguousInventory$' -count=1 -v` passed all 9 subtests in 3.475s. The uncached full package `go test -count=1 ./scripts/tests/refresh-flux-ghcr-auth/...` also passed in 177.172s. Independently, the Linux `🔐 Validate Flux GHCR credential bridge` step in Actions run `33918714791` completed successfully at 2026-09-04T21:02:31Z. The claimed missing-log ENOENT state therefore did not reproduce in either the local exact test or the workflow environment. I am leaving `readLines` strict so a genuinely missing operation log remains a test failure rather than being silently treated as empty.
